include/openssl/evp.h

```diff
@@ -1704,9 +1704,14 @@ const char *EVP_PKEY_get0_type_name(const EVP_PKEY *key);
 # define EVP_PKEY_OP_ENCAPSULATE         (1<<12)
 # define EVP_PKEY_OP_DECAPSULATE         (1<<13)
 
-# define EVP_PKEY_OP_TYPE_SIG    \
-        (EVP_PKEY_OP_SIGN | EVP_PKEY_OP_VERIFY | EVP_PKEY_OP_VERIFYRECOVER \
-                | EVP_PKEY_OP_SIGNCTX | EVP_PKEY_OP_VERIFYCTX)
+# define EVP_PKEY_OP_SIGNMSG             (1<<14)
+# define EVP_PKEY_OP_VERIFYMSG           (1<<15)
+
+# define EVP_PKEY_OP_TYPE_SIG                           \
+        (EVP_PKEY_OP_SIGN | EVP_PKEY_OP_SIGNMSG         \
+         | EVP_PKEY_OP_VERIFY | EVP_PKEY_OP_VERIFYMSG   \
+         | EVP_PKEY_OP_VERIFYRECOVER                    \
+         | EVP_PKEY_OP_SIGNCTX | EVP_PKEY_OP_VERIFYCTX)
```

Comment:
Agreed.  Generally speaking, that's a functionality set that's missing more broadly than just for the functionality of this PR, probably because we took "know your provider" a bit too far.  Anyway, "can-do" functionality probably deserves a PR of its own.

Reply:
Yes, it can be added later. In this particular case it is more useful than in other cases, but yeah, could be made generic. Something like: `EVP_SIGNATURE_supports(signature, EVP_PKEY_OP_SIGNMSG)`

Reply:
Oh!  That was a good example that doesn't reinvent too much